compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -597,6 +598,72 @@ struct DistributeScfFor final : OpDistributionPattern<scf::ForOp> {
   }
 };
 
+struct DistributeBroadcastLayoutAttr final
+    : OpDistributionPattern<vector::BroadcastOp> {
+  using OpDistributionPattern::OpDistributionPattern;
+
+  LogicalResult matchAndRewrite(vector::BroadcastOp broadcastOp,
+                                DistributionSignature &signature,
+                                PatternRewriter &rewriter) const override {
+
+    VectorValue source = dyn_cast<VectorValue>(broadcastOp.getSource());
+    if (!source) {
+      // TODO: Add support for scalar broadcasting.
+      llvm::errs() << "scalar??\n";
+      return failure();
+    }
+
+    VectorValue vector = broadcastOp.getVector();
+    LayoutAttr layout = dyn_cast<LayoutAttr>(signature[vector]);
+    if (!layout) {
+      llvm::errs() << "layout is not layoutattr\n";
+      return failure();
+    }
+
+    VectorLayoutInterface sourceLayout = signature[source];
+
+    // We currently only support 1-D to 2-D broadcasting.
+    if (source.getType().getRank() != 1 || vector.getType().getRank() != 2) {
+      llvm::errs() << "rank mismatch\n";
+      return failure();
+    }
+
+    int broadcastedDim = 0;
```

Comment:
I think there are some assumptions here about the 1-D value. Specifically, the assumption is that the 1-D value has been obtained by reducing a 2-D value and that the reduction dimension is 0 and the parallel dimension is 1. If so, we should add checks for that. What do you think?

Reply:
I think there is a check for this just above:

```
    // We currently only support 1-D to 2-D broadcasting.
    if (source.getType().getRank() != 1 || vector.getType().getRank() != 2) {
      llvm::errs() << "rank mismatch\n";
      return failure();
    }
```

Reply:
This checks that the source Is 1d and result 2d. But below there is an assumption that parallel dim = 1 and reduction dim = 0, right? I guess what I am asking is - is it possible for parallel dim =0 and reduction dim = 1 or will it always be the case that parallel dim = 1 and reduction dim = 0?

Reply:
Okay so seems like vector.broadcast will always broadcast dim 0. Thanks for clarifying.